src/learning/modules.py

```diff
@@ -0,0 +1,113 @@
+import torch.nn as nn
+import utils
+
+
+class CycleGANLoss(nn.Module):
+    def __init__(self, regularizer, inverted, device):
+        super(CycleGANLoss, self).__init__()
+        self.regularizer = regularizer
+        self.inverted = inverted
+        self.device = device
+
+    def forward(self, data):
+        mse = nn.MSELoss(reduction='elementwise_mean').to(self.device)
+        l1norm = nn.L1Loss(reduction='elementwise_mean').to(self.device)
+
+        batch_targets = utils.get_target(
+            True, self.inverted, data.batch_x_predictions.size(), self.device)
+        synthesis_targets = utils.get_target(
+            False, self.inverted, data.synthesis_x_predictions.size(),
+            self.device)
+
+        return mse(data.batch_x_predictions, batch_targets) +\
+            mse(data.synthesis_x_predictions, synthesis_targets) +\
+            mse(data.batch_y_predictions, batch_targets) +\
+            mse(data.synthesis_y_predictions, synthesis_targets) +\
+            self.regularizer * l1norm(data.id_x_approximations,
+                                      data.batch_x) +\
+            self.regularizer * l1norm(data.id_y_approximations,
+                                      data.batch_y)
+
+
+class ResidualBlock(nn.Module):
+    def __init__(self, n_channels, kernel_size, use_dropout):
+        super(ResidualBlock, self).__init__()
+        block_1 = [nn.ReflectionPad2d(1),
+                   nn.Conv2d(n_channels, n_channels, kernel_size, padding=0),
```

Comment:
When we have a Conv2d or Linear layer followed by a InstanceNorm or BatchNorm layer, we can set the bias of the Conv layer to false, since the bias term is taken care of in the normalisation layer: https://github.com/KaimingHe/deep-residual-networks/issues/10#issuecomment-194037195.
Will save us some parameters.
```suggestion
                   nn.Conv2d(n_channels, n_channels, kernel_size, padding=0, bias=False),
```

Reply:
Sweet! :)